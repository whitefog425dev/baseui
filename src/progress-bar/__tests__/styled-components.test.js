/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {StyledRoot, StyledBar, StyledBarProgress, StyledLabel} from '../index';

describe('ProgressBar styled components', () => {
  // $FlowFixMe
  describe.each([
    [StyledRoot, 'StyledRoot'],
    [StyledBar, 'StyledBar'],
    [StyledBarProgress, 'StyledBarProgress'],
    [StyledLabel, 'StyledLabel'],
  ])('Test %% %s', (Component, compName) => {
    test(compName, () => {
      const props = {
        $value: 40,
        $successValue: 100,
      };
      const component = shallow(
        <Component {...props}>
          <div />
        </Component>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        `${compName} has correct styles`,
      );
    });
  });
});
