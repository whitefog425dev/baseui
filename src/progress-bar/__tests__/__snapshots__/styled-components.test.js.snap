// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar styled components Test % function MockStyledComponent() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, MockStyledComponent);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = MockStyledComponent.__proto__ || Object.getPrototypeOf(MockStyledComponent)).call.apply(_ref, [this].concat(args))), _this), _this.state = {}, _temp), _possibleConstructorReturn(_this, _ret);
    } StyledBar StyledBar: StyledBar has correct styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.mono400",
  "borderRadius": "$theme.sizing.scale0",
  "height": "4px",
  "marginBottom": "$theme.sizing.scale500",
  "marginLeft": "$theme.sizing.scale500",
  "marginRight": "$theme.sizing.scale500",
  "marginTop": "$theme.sizing.scale500",
}
`;

exports[`ProgressBar styled components Test % function MockStyledComponent() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, MockStyledComponent);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = MockStyledComponent.__proto__ || Object.getPrototypeOf(MockStyledComponent)).call.apply(_ref, [this].concat(args))), _this), _this.state = {}, _temp), _possibleConstructorReturn(_this, _ret);
    } StyledBarProgress StyledBarProgress: StyledBarProgress has correct styles 1`] = `
Object {
  "backgroundColor": "$theme.colors.primary400",
  "borderRadius": "$theme.sizing.scale0",
  "height": "100%",
  "transition": "width 0.5s",
  "width": "40%",
}
`;

exports[`ProgressBar styled components Test % function MockStyledComponent() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, MockStyledComponent);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = MockStyledComponent.__proto__ || Object.getPrototypeOf(MockStyledComponent)).call.apply(_ref, [this].concat(args))), _this), _this.state = {}, _temp), _possibleConstructorReturn(_this, _ret);
    } StyledLabel StyledLabel: StyledLabel has correct styles 1`] = `
Object {
  "color": "$theme.colors.mono700",
  "fontFamily": "$theme.typography.font250.fontFamily",
  "fontSize": "$theme.typography.font250.fontSize",
  "fontWeight": "$theme.typography.font250.fontWeight",
  "lineHeight": "$theme.typography.font250.lineHeight",
  "textAlign": "center",
}
`;

exports[`ProgressBar styled components Test % function MockStyledComponent() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, MockStyledComponent);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = MockStyledComponent.__proto__ || Object.getPrototypeOf(MockStyledComponent)).call.apply(_ref, [this].concat(args))), _this), _this.state = {}, _temp), _possibleConstructorReturn(_this, _ret);
    } StyledRoot StyledRoot: StyledRoot has correct styles 1`] = `
Object {
  "width": "100%",
}
`;
